### PR TITLE
Order product employee service logic

### PR DESCRIPTION
**Pull request by mantvydasMac**
Order, Product, Employee model updates and service methods. It's not finished yet. Order things are mostly done, but Product and Employee service methods still need changes, validation checks and other things. Product models updated to use proper annotation. The methods will be finished when working on the entire order management flow.

### CODE REVIEW

**Review by MartoLedas — APPROVED** [review]
ok

File: src/main/java/com/example/wdmsystem/employee/system/EmployeeController.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ ResponseEntity<Employee> getEmployee(@PathVariable int employeeId) {
     @PutMapping("/employees/{employeeId}")
     ResponseEntity<Employee> updateEmployee(@PathVariable int employeeId, @RequestBody EmployeeDTO request) {
         _employeeService.updateEmployee(employeeId, request);
-        return new ResponseEntity<>(null, HttpStatus.OK);
+        return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
```
Comment (AkvileR) [inline src/main/java/com/example/wdmsystem/employee/system/EmployeeController.java:42]: This change deviates from the yaml file, so make sure to mark it down somehow to later put in the "what we changed" file.

File: src/main/java/com/example/wdmsystem/order/system/OrderDTO.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.example.wdmsystem.order.system;
 
-public record OrderDTO() {
+import java.time.LocalDateTime;
+
+public record OrderDTO(Integer id, OrderStatus status) {
```
Comment (AkvileR) [inline src/main/java/com/example/wdmsystem/order/system/OrderDTO.java:5]: Why did we change the OrderDTO? The yaml defines a perfectly good dto (well if we take out the total price part), you wouldn't need to change the createOrder() parameters then.

File: src/main/java/com/example/wdmsystem/product/system/ProductController.java [hunk 3]
```diff
@@ -24,13 +24,13 @@ public ProductController(ProductService productService) {
     }
 
     @PostMapping("/products")
-    ResponseEntity<Product> createProduct(@RequestBody ProductDTO request) {
+    ResponseEntity<Product> createProduct(@RequestBody Product request) {
```
Comment (AkvileR) [inline src/main/java/com/example/wdmsystem/product/system/ProductController.java:27]: Are you sure we should pass the entire product entity and not a dto? The id, for example, is generated later, createdAt is also assigned later, do we really need to expose and pass those values from the front end?

**Review by aWhiteScreen — APPROVED** [review]
approved

**Pull request merged**